### ISSUE TITLE
Error "Platform innova does not generate unique IDs." with multiple devices.

### ISSUE THREAD

**Issue by hendrikdonvil** [danielrivard/homeassistant-innova#131]
I have an issue connecting my second Innova 2.0 device. It appears to be because both have the same serial number, and it is used by your integration to create the device ID in Home Assistant. 

I saw in my home assistant logs these errors:
Platform innova does not generate unique IDs. ID INxxxxxxx-scheduling-switch already exists - ignoring switch.inxxxxxxx_scheduling_switch

In the Innova rest API, I noticed that for both devices, the 'serial' field is the same. 

I've seen you default to the serial id in device_info.py:
```
self._unique_id = self._innova.serial
if not self._unique_id:
        self._unique_id = self._innova.uid
```
Maybe it's a good idea to invert that, and use the uid as a default? That would probably make everything work for me.

Thanks for considering this!

Hendrik

**Comment by hendrikdonvil** [comment]
I'm going to close this issue. Apparently, I 'chose' the serial numbers myself when setting up the innova units. Not what I expected from a serial number. I reset my device, chose another serial number, and it works now.

**Comment by danielrivard** [comment]
Hi, yes, their setup is a bit weird, the serial number should not be changeable by the user.

The UID is actually the MAC address, so I preferred to use the serial over it as it seemed to be the better choice. But that was before I realized that the serial was changeable by the user.

I'll still check and test what would happen if I used the UID first. Because I used the serial already, I am afraid it could break the installation for existing users as the unique id is used to store the device in Home Assistant's DB.
